modules/transaction-payment/src/lib.rs

```diff
@@ -825,31 +821,101 @@ where
 					<T as Config>::Currency::ensure_can_withdraw(who, fee, reason, new_free_balance).is_ok()
```

Comment:
why do we need total balance and free balance check and `ensure_can_withdraw`?
I think `ensure_can_withdraw` by itself is enough?

Reply:
`ensure_can_withdraw` refer to substrate frame `Balances`: https://github.com/paritytech/substrate/blob/f5f286db0da99761792b69bf4a997535dcc8f260/frame/balances/src/lib.rs#L1473-L1484

I think the simpliest way is just compare `fee + ed` with `free`. if fee+ed>free, native is not enough, result=`fee+ed-free`.  The reason we need `-free` is for newly user, first tx, native is not enough, result=`fee+ed-0`, after withdraw fee, free=`ed`.  second tx, native is still not enough, result=`fee+ed-free`=fee.